python/packages/autogen-ext/src/autogen_ext/teams/magentic_one.py

```diff
@@ -36,12 +42,48 @@ async def example_usage():
 
             if __name__ == "__main__":
                 asyncio.run(example_usage())
+
+    
+        .. code-block:: python
+
+            # Enable human-in-the-loop mode        
+            import asyncio
+            from autogen_ext.models.openai import OpenAIChatCompletionClient
+            from autogen_ext.teams.magentic_one import MagenticOne
+            from autogen_agentchat.ui import Console
+
+
+            async def example_usage_hil():
+                client = OpenAIChatCompletionClient(model="gpt-4o")
+                # to enable human-in-the-loop mode, set hil_mode=True
+                m1 = MagenticOne(client=client, hil_mode=True)
+                task = "Write a Python script to fetch data from an API."
+                result = await Console(m1.run_stream(task=task))
+                print(result)
+
+
+            if __name__ == "__main__":
+                asyncio.run(example_usage_hil())
+
+    References:
+        .. code-block:: bibtex
+
+            @article{fourney2024magentic,
+                title={Magentic-one: A generalist multi-agent system for solving complex tasks},
+                author={Fourney, Adam and Bansal, Gagan and Mozannar, Hussein and Tan, Cheng and Salinas, Eduardo and Niedtner, Friederike and Proebsting, Grace and Bassman, Griffin and Gerrits, Jack and Alber, Jacob and others},
+                journal={arXiv preprint arXiv:2411.04468},
+                year={2024}
+            }
     """
 
-    def __init__(self, client: OpenAIChatCompletionClient):
+    def __init__(self, client: OpenAIChatCompletionClient, hil_mode: bool = False):
```

Comment:
Use `autogen_core.models.ChatCompletionClient` base class instead of concrete class for the argument type

Reply:
I did this because we only tested with openai

Reply:
Ah I see. Though this client has already been used by many for Ollama models. Also there is AzureOpenAIChatCompletionClient, which we also tested on. 

I think we need to validate the model capabilities in the constructor -- unless it has already been done in the base class. Sorry on mobile so a bit hard to switch page here. 

We can also validate the model name, and raise warning if the model is something we haven't tested.

Reply:
Okay good point. I will do this.

Reply:
Done.
1. I added a private method to the class that validates the client.
   <img width="806" alt="image" src="https://github.com/user-attachments/assets/d0542001-901c-408e-ae2a-335c39b8cb27" />


2. I had to expose the `_create_args` in `BaseOpenAIChatCompletion` to achieve this.
    <img width="598" alt="image" src="https://github.com/user-attachments/assets/d96891d7-dff8-4897-ac71-0ed8a6b28168" />

Reply:
@jackgerrits what do you think? 

Perhaps instead of exposing the create_args, we can just have a property for `model`.

Reply:
Let's not change the interface for this change - we're planning on introducing "model family" which will fix this. I would say for this PR just do an isinstance check for openai or azure openai and dig into the internals and check the model field